05_stringer/nickolee/stringer_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := "127.0.0.1"
+	actual := buf.String()
+
+	fmt.Println("expected: ", expected)
+	fmt.Println("actual: ", actual)
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestStringMethod(t *testing.T) {
+	type test struct {
+		name     string
+		input    IPAddr
+		expected string
+	}
+
+	tests := []test{
+		{name: "all local IPv4 addresses", input: IPAddr{0, 0, 0, 0}, expected: "0.0.0.0"},
+		{name: "google dns/a lucky number :)", input: IPAddr{8, 8, 8, 8}, expected: "8.8.8.8"},
+		{name: "localhost/loopback", input: IPAddr{127, 0, 0, 1}, expected: "127.0.0.1"},
+		{name: "wikipedia", input: IPAddr{208, 80, 154, 224}, expected: "208.80.154.224"},
+		{name: "facebook", input: IPAddr{31, 13, 71, 36}, expected: "31.13.71.36"},
+		{name: "nil value", input: IPAddr{}, expected: "0.0.0.0"}, // defaults to the nil value of a byte
+		{name: "missing", input: IPAddr{192, 150}, expected: "192.150.0.0"},
+	}
+
+	for _, testCase := range tests {
+		actual := testCase.input.String()
+		fmt.Println(testCase.name, "expected: ", testCase.expected)
```

Comment:
I guess it is up to you and what you want your tests to display, but it goes against normal convention to have print statements inside your tests like this. Normally you would only want something to be printed or logged when there is an error (if all the tests are passing / ok, you don't care what the expected / actual was for every test). This comes out of the box anyway with the `t.Fatal` so you don't really need to include anything extra

Reply:
Please fix, no debug messages in tests.

Reply:
Thanks have noted and fixed this